README.md

```diff
@@ -133,6 +137,75 @@ Assets and events will ignore existing ids on deletes. If you prefer to abort th
 when an attempt to delete an unknown id, use `.option("ignoreUnknownIds", "true")`
 for those resources types.
 
+## Asset hierarchy (beta)
+Note: The asset hierarchy utility is currently in beta, and has not been sufficiently tested to be used 
+on production data.
+
+The `.option("type", "assethierarchy")` lets you write new asset hierarchies, or update existing ones,
+using the Spark data source. It currently supports writing a single root asset and any number of children.
+The asset hierarchy utility can ingest entire trees, as long as all nodes are connected to the root asset
+through the `externalId`/`parentExternalId` relationship. If input contains an update to data that already exists,
+i.e there's a match on `externalId` and there's a change to one of the other fields, the asset will be updated.
+There's also an option to delete assets from CDF that are not referenced in the input data.
+
+### Requirements
+- The source data must contain a single root, denoted by setting its `parentExternalId` to the empty string `""`.
+- All assets, except root, must be connected to another asset in the input data by `parentExternalId`.
+- The input data must not have loops, to ensure all asset trees are fully connected.
```

Comment:
Are we doing anything to detect cycles?

Reply:
We are requiring all assets to be connected to the root, which is not connected to anything, which will discover cycles (at least I can't think of a counter-example)

Reply:
That's true, good point 👍